src/org/exist/xquery/parser/XQueryTreeParser.java

```diff
@@ -80,7 +81,7 @@ public boolean foundErrors() {
 	}
 
 	public String getErrorMessage() {
-		StringBuilder buf= new StringBuilder();
+		StringBuffer buf= new StringBuffer();
```

Comment:
This was in the Antlr `XQueryTree.g` grammar which generates this Java file, I didn't change this previously. However, I just fixed it in a subsequent commit for you